src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/Tasks/TaskHelper.cs

```diff
@@ -318,6 +316,8 @@ internal static bool CouldReferenceAssemblyBeChanged(string assemblyPath, string
                         }
                     }
                 }
+#if NETFX
```

Comment:
Can this be refactored into a separate method, so that `#if NETFX` is not necessary inside the body of `CouldReferenceAssemblyBeChanged` ? 

Something like this: 

```C#

internal static bool CouldReferenceAssemblyBeChanged(...)
{
   ...
    // Check the Global Assembly Cache (GAC) on .NET Framework only.
    TestGacPaths(ref bCouldBeChanged); 
   ...
}


private static void TestGacPaths(ref bool bCouldBeChanged)
{
#if NETFX
    IEnumerable<string> gacRoots = GetGacRoots();
    if (gacRoots != null)
    {
        foreach(var gacRoot in gacRoots)
        {
            if (.... )
            {
                bCouldBeChanged = false;
            }
        }
    }
#endif 
}

```